### ISSUE TITLE
There should be a zero(::Missing) method

### ISSUE THREAD

**Issue by yurivish** [JuliaLang/julia#28854]
`zero(Missing)` currently fails, as does `one(Missing)` and `oneunit(Missing)`:

```julia
julia> zero(Missing)
ERROR: UndefVarError: T not defined
```

The error comes from this code in base/missing.jl:
```julia
for f in (:(Base.zero), :(Base.one), :(Base.oneunit))
    @eval function $(f)(::Type{Union{T, Missing}}) where T
        T === Any && throw(MethodError($f, (Any,)))  # To prevent StackOverflowError
        $f(T)
    end
end
```

Perhaps `zero(::Missing)` and `one(::Missing)` should be `missing`, since `missing + missing == missing` and the same for `*` and those two methods give you the additive and multiplicative identities.

@ararslan suggests adding `@eval $(f)(::Type{Missing}) = missing` to the loop.

**Comment by nalimilan** [comment]
Good question. It sounds a bit weird that `iszero(zero(missing))` and `zero(missing) == 0` would not be `true`. But maybe that's not an issue. Can you show an actual use case?

**Comment by yha** [comment]
I just encountered this when doing correlation plots (from `StatsPlots`). 
It seems to arise easily by calling `Statistics.cor` on data with `missing`. Reproducing example:
```
julia> import Statistics: cor
julia> cor([zeros(1,1); missing])
ERROR: UndefVarError: T not defined
```

**Comment by nalimilan** [comment]
This has actually been fixed in March by https://github.com/JuliaLang/julia/pull/31303. That will be included in Julia 1.2.
